src/neptune_api/client.py

```diff
@@ -394,3 +399,103 @@ def sync_auth_flow(self, request: httpx.Request) -> Generator[httpx.Request, htt
     async def async_auth_flow(self, request: httpx.Request) -> typing.AsyncGenerator[httpx.Request, httpx.Response]:
         # TODO: Missing implementation
         yield request
+
+
+USE_IAP = os.environ.get("NEPTUNE_GCP_USE_IAP", False) in {"true", "t", "1", "True"}
+
+if USE_IAP:
+    import google.auth
+    from google.cloud import iam_credentials_v1
+
+    NEPTUNE_IAP_SERVICE_ACCOUNT = os.environ["NEPTUNE_IAP_SERVICE_ACCOUNT"]
+
+    class IdentityAwareProxyAuthenticator(httpx.Auth):
```

Comment:
I think it's better to move this class to a separate module and import selectively when env var is enabled. Or even use sth like this as the `auth` argument to `Client` above:
```
def make_authenticator():
   if not iap: return standard_auth_func
   else: return <imported google auth>
```
Point is: keep everything vendor-specific in a separate submodule